app/src/main/java/com/example/indoor_volleyball/QueryActivity.java

```diff
@@ -0,0 +1,93 @@
+package com.example.indoor_volleyball;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Toast;
+
+import com.example.indoor_volleyball.Models.Event;
+import com.example.indoor_volleyball.Models.Gym;
+import com.example.indoor_volleyball.databinding.ActivityQueryBinding;
+import com.parse.FindCallback;
+import com.parse.ParseException;
+import com.parse.ParseGeoPoint;
+import com.parse.ParseQuery;
+import com.parse.ParseUser;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class QueryActivity extends AppCompatActivity {
+
+    private static final String TAG = "QueryActivity";
+    private ActivityQueryBinding binding;
+    List<Gym> allGyms;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_query);
+        allGyms = new ArrayList<>();
+
+        binding = ActivityQueryBinding.inflate(getLayoutInflater());
+
+        View view = binding.getRoot();
+
+        setContentView(view);
+
+        binding.tvQuery.setText("");
+        queryAllGyms();
+
+        binding.btAllGymsList.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                for (int i =0; i<allGyms.size();i++) {
```

Comment:
general tip, but you can use a foreach loop if you don't need to have index numbers and you just need to access each object.  It looks like:

`foreach(Gym gym : allGyms) {
      gym.doSomething }
`
In english it's like "for each Gym object in the allGyms list, do me a loop and call each object gym."
https://www.geeksforgeeks.org/for-each-loop-in-java/